### PR TITLE
Handle ReviveDBRestorePointNotFound error

### PR DESCRIPTION
**Pull request by jizhuoyu**
- operator now handles ReviveDBRestorePointNotFound error returned from vclusterops
- e2e test for invalid id and index
- e2e test for cluster scaling before restoring

### CODE REVIEW

File: go.sum
Comment (spilchen) [inline go.sum:323]: Make sure you do a go mod tidy. This will clean out the old vcluster package.

File: tests/e2e-leg-6/revivedb-1-node-restore/17-assert.yaml
Comment (spilchen) [inline tests/e2e-leg-6/revivedb-1-node-restore/17-assert.yaml:1]: I like that you added scaling to this test, but is it a 1-node revive anymore? Maybe revivedb-scaling-restore?

Comment (jizhuoyu) [reply tests/e2e-leg-6/revivedb-1-node-restore/17-assert.yaml:1]: done.

File: go.sum
Comment (jizhuoyu) [reply go.sum:321]: done.

**Comment by chinhtranvan** [comment]
I don't have any comments in your PR. e2e leg 6 was failed because it also has log rotation for 'editor.log.' I just fix the test in my PR. Perhaps, you can wait until my PR is merged.

**Review by spilchen — APPROVED** [review]

**Pull request merged**
